src/sidebar/store/modules/annotations.js

```diff
@@ -551,12 +576,13 @@ const orphanCount = createSelector(
 /**
  * Return all loaded annotations which have been saved to the server
  *
- * @return {Annotation[]}
+ * @param {State} state
+ * @return {SavedAnnotation[]}
  */
 function savedAnnotations(state) {
-  return state.annotations.filter(ann => {
-    return !metadata.isNew(ann);
-  });
+  return /** @type {SavedAnnotation[]} */ (
+    state.annotations.filter(ann => isSaved(ann))
+  );
```

Comment:
I'm curious why the extra typing is needed here.

Reply:
The immediate answer was because TS failed to figured out the effect of `filter` automatically. Upon further experimentation, I found that this works without additional type annotations:

```js
return state.annotations.filter(isSaved);
```

The issue seems to be that the `ann => isSaved(ann)` callback loses the nuance of the `isSaved` return type and treats it as just `boolean`.

The problem with this workaround is that if `isSaved` ever gains support for additional parameters, this could have unexpected interactions with `filter`.